src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -376,6 +376,11 @@ public unsafe void SetResponseHeaders()
                 var knownHeaderIndex = HttpApiTypes.HTTP_RESPONSE_HEADER_ID.IndexOfKnownHeader(headerPair.Key);
                 for (var i = 0; i < headerValues.Count; i++)
                 {
+                    if(string.IsNullOrEmpty(headerValues.ToString()))
+                    {
+                        throw new ArgumentException($"{headerPair.Key} header should be non-empty string");
```

Comment:
In other servers we skip the empty values wherever plausible. It's pretty safe to ignore empty headers.
https://github.com/aspnet/AspNetCore/blob/fad6a1867ccb42bcf23c8507218b944e8798a06a/src/Servers/HttpSys/src/RequestProcessing/Response.cs#L477-L480
Or specify an empty value when it can work.
https://github.com/aspnet/AspNetCore/blob/fad6a1867ccb42bcf23c8507218b944e8798a06a/src/Servers/HttpSys/src/RequestProcessing/Response.cs#L507

Reply:
Being consistent across servers sounds good :)